Assets/Scripts/GameManagement/DataPersistence/FileDataHandler.cs

```diff
@@ -84,6 +86,41 @@ public object ReadObjectFromJson(Type type)
         }
     }
 
+    private string GetSingleJsonObjectString(string tag)
+    {
+        string array = "";
+        string json = fileReader.ReadToEnd();
+        int startIndex = json.IndexOf(tag);
```

Comment:
Just realized, this could cause problems if, for example, Inventory had a field called "respawnPoint" and was in the JSON before the respawnPoint storage, then you tried to load respawnPoint. Doesn't cause problems yet but could be in the future, might be a good reason to use C#'s JSON serializer.